WordPress/src/main/java/org/wordpress/android/WordPress.java

```diff
@@ -763,6 +760,9 @@ public void onTrimMemory(final int level) {
                     break;
                 case TRIM_MEMORY_BACKGROUND:
                 case TRIM_MEMORY_UI_HIDDEN:
+                    // See https://android.jlelse.eu/how-to-detect-android-application-open-and-close-background-and-foreground-events-1b4713784b57
+                    onAppGoesToBackground();
```

Comment:
We need to check the level here don't we?

```kotlin
if (level == ComponentCallbacks2.TRIM_MEMORY_UI_HIDDEN) {
    lifecycleListener.onAppGoesToBackground()
}
```

Reply:
Oh maybe? Its seemed fine to for TRIM_MEMORY_BACKGROUND and TRIM_MEMORY_UI_HIDDEN events so far. At least I'm not seeing the `onAppGoesToBackground` called multiple times.  For paranoia I can make it more restrictive.

Reply:
I don't remember a lot about this, but IIRC both `TRIM_MEMORY_BACKGROUND` and `TRIM_MEMORY_UI_HIDDEN` might be overly permissive and in low-memory situations you might end up having both of them called.